node/tests/GlideClusterClient.test.ts

```diff
@@ -1966,4 +1966,316 @@ describe("GlideClusterClient", () => {
             }
         },
     );
+    describe("GlideClusterClient - AZAffinity Read Strategy Test", () => {
+        it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+            "should route GET commands to all replicas with the same AZ using protocol %p",
+            async (protocol) => {
+                const az = "us-east-1a";
+                const n_replicas = 6;
+                const GET_CALLS = 3 * n_replicas;
+                const get_cmdstat = `calls=${GET_CALLS / n_replicas}`;
+
+                let client_for_config_set;
+                let client_for_testing_az;
+
+                try {
+                    // Stage 1: Configure nodes
+                    client_for_config_set =
+                        await GlideClusterClient.createClient(
+                            getClientConfigurationOption(
+                                cluster.getAddresses(),
+                                protocol,
+                            ),
+                        );
+
+                    // Skip test if version is below 8.0.0
+                    if (cluster.checkIfServerVersionLessThan("8.0.0")) {
```

Comment:
Why @adarovadya uses lt 7.2 and you 8?

Reply:
Yesterday, when I was implementing the code, I had Valkey 7.2.5 and it did not have the availability_zone field when I ran INFO.
Solved the issue when I updated it to 8.0.1
I will look up/test the first version that supports availability_zone field

Reply:
Yes i understod that ValKey is supporting from 8.0, EC supporting from 7.2, so thats fine.
Adar is changing code as well.